### DISCUSSION TITLE
Does AGH resort to TCP?

### DISCUSSION THREAD

**Discussion by agneevX** [AdguardTeam/AdGuardHome#4372] [General]
I was wondering when using the "default" upstream scheme of `[IP][:PORT]`, it's UDP by default, but does it fallback to TCP for larger payloads?

If so, I assume #4166 is UDP only?

**Comment by ainar-g** [comment]
Apologies for the late response.  Yes, when AGH receives a truncated response, it retries the request using TCP.
